### DISCUSSION TITLE
How to get scope for comment?

### DISCUSSION THREAD

**Discussion by DanielSWolf** [eslint/eslint#15691] [Q&A] [answered]
Consider the following code:

```ts
/** Prints a {@link Person}. */
function printPerson(person: Person) {
  console.log(person.name);
}

interface Person {
  name: string;
  age: number;
}
```

I'm using `@typescript-eslint/parser` as parser.

I'm writing a rule that validates `@link` tags, making sure that they only reference symbols that are actually in scope. So I need to determine the scope of the comment in line 1.

Currently, I'm using `sourceCode.getAllComments()` to get a list of all comments in the code. That works, but `sourceCode.scopeManager.acquire(commentNode)` always returns `null`. So I can't get the scope for a given comment node.

Is there any way of determining which comment belongs to which code node and thus has which scope?

**Comment by nzakas** [comment, marked as answer]
Comments aren’t considered to be part of a scope, so this won’t work. Your best bet is to look for function declarations that are preceded by comments and determine the scope by using the function declaration

**Reply by DanielSWolf** [reply]
Makes sense. I'll search for the first node whose location is after that of the comment. Thanks!
